tools/scripts/autorun_generateSoundMetadata.rb

```diff
@@ -0,0 +1,123 @@
+# Use this script for converting multiple audio formats into .mp3 (required)
+# Then generateSoundMetadata.rb will be run to:
+# 1. create metadata (JSON) for each file (required)
+# 2. use file_path to get the directory name like animals or notifications
+# to add category_animals or category_notifications to the aliases (required)
+
+# command to run file: bundle exec ./autorun_generateSoundMetadata.rb [folder_path] [csv_location]
+# example for folder_name = Alert
+# example for csv_location = ./sound_metadata.csv
+# example command = bundle exec ./parse_new_sound_aliases_csv.rb ~/Downloads/Animals ./sound_metadata.csv
+require 'csv'
+require 'fileutils'
+
+class NewSoundAliasesMetadataBuilder
+  def initialize(csv)
+    @csv = CSV.parse(File.read(csv), headers: true)
+  end
+
+  def fix_corrupted_filename(name, type_of_file)
+    name.downcase! if name =~ /[A-Z]/
+    # some filenames had spaces instead of underscores
+    name.tr!(" ", "_") if name.include?(" ")
+    # some filenames have more than one underscore but we want to match
+    #  the style of existing sound filenames with one underscore
+    name.gsub!(/_{2,}/, "_") if name =~ /_{2,}/
+    name.gsub!(/^\_+/, '') if name =~ /^\_+/
+
+    if name.chomp([-4..-1].to_s) =~ /\_+$/
+      name.gsub!(/\_+$/, '')
+      name.chomp!([-4..-1].to_s)
+      name << ".mp3" if type_of_file == "csv"
+    end
+
+    return name
+  end
+
+  def convert_wav_to_mp3(folder_path)
+    # converts .wav files to .mp3
+    puts %x(
+        for file in "#{folder_path}"/*.wav
+        do \`ffmpeg -i "$file" "${file%.wav}".mp3\`
+          done
+      )
+    Dir.mkdir "#{folder_path}/mp3_files"
+
+    Dir[folder_path + "/*.mp3"].each do |f|
+      filename = File.basename(f, File.extname(f))
+
+      if filename =~ /(^\_+|\_+$|_{2,}|\s|[A-Z])/
+        fixed_name = fix_corrupted_filename(filename, "audio")
+        File.rename(f, folder_path + "/" + fixed_name + File.extname(f))
+        f.gsub!(f.split('/')[-1].to_s, "#{fixed_name}.mp3")
+      end
+
+      FileUtils.mv(f.to_s, "#{folder_path}/mp3_files")
+    end
+  end
+
+  def parse_filenames_in_csv(csv)
+    filename = csv.by_col[1]
+    # changed file extension to .mp3 to run .generateSoundMetaData.rb
+    return filename.each do |name|
+      # original file extensions are .wav but they shoud be .mp3
+      # in order to run ./generateSoundMetaData.rb
+      name.gsub!(".wav", ".mp3")
+
+      if name =~ /(^\_+|\_+$|_{2,}|\s|[A-Z])/
+        fix_corrupted_filename(name, "csv")
+      end
+    end
+  end
+
+  def parse_aliases_in_csv(csv)
+    # removed spaces to simplify
+    # because csv file had aliases that had
+    # more than one space or none
+    aliases_with_spaces_removed = csv.by_col[3].each do |aliase|
+      aliase.gsub!(/,\s+/, ',').downcase!
+      aliase.tr!(" ", ',')
+    end
+
+    return aliases_with_spaces_removed.map do |aliase|
+      "\"#{aliase.split(/\,/).join('\', \'').gsub!(/,\s+/, ',')}\""
```

Comment:
If I'm reading this correctly, you've got double-quotes on the outside and single-quotes on the inside, but no outer single-quotes... so I'm a bit confused by the format.

Trying this out in `irb`:
```
irb(main):001:0> aliase = 'air,fire,water'
=> "air,fire,water"
irb(main):002:0> "\"#{aliase.split(/\,/).join('\', \'').gsub!(/,\s+/, ',')}\""
=> "\"air','fire','water\""
```

Or am I making a bad assumption about the initial input?  I suppose if the aliases are already single-quoted collectively you get this:
```
irb(main):003:0> aliase = "\'air,fire,water\'"
=> "'air,fire,water'"
irb(main):004:0> "\"#{aliase.split(/\,/).join('\', \'').gsub!(/,\s+/, ',')}\""
=> "\"'air','fire','water'\""
```

I also think the `gsub!` call here is redundant with the one a few lines above, and could be removed.

Reply:
I believe I missed changing the 2nd double quote in `"\"air'` to a single quote after changing the double quotes in `.join('\', \'')` to single quotes when the rubocop linter indicated that it prefers single-quoted strings inside interpolations. The script continued to run and generate appropriate aliases so I carelessly missed it again! 

In my attempt to change to double quote, I found out that generateSoundMetadata doesn't need the aliases to be written as `"hello","world"` which is why I formatted the them that way  but `"hello,world"` is suffice to create 2 aliases `"hello", "world"` so I ended up removing the double quotes and just kept the aliases as `"hello,world,yay"`.